piccolo/columns/column_types.py

```diff
@@ -77,6 +80,23 @@ class Band(Table):
     from piccolo.table import Table
 
 
+class Arguments(t.TypedDict, total=False):
+    null: NotRequired[bool]
+    primary_key: NotRequired[bool]
+    primary: NotRequired[bool]
+    key: NotRequired[bool]
+    unique: NotRequired[bool]
+    index: NotRequired[bool]
+    index_method: NotRequired[IndexMethod]
+    required: NotRequired[bool]
+    help_text: NotRequired[str]
+    choices: NotRequired[t.Type[Enum]]
+    secret: NotRequired[bool]
+    auto_update: NotRequired[t.Any]
+    digits: NotRequired[t.Tuple[int, int]]
+    db_column_name: NotRequired[str]
```

Comment:
Do you need `NotRequired`? As you've specified `total=False`?

Reply:
> I am very optimistic about this Unpack approach. It's not just useful for autocompletion - VSCode, MyPy etc will detect typos. We might be able to remove the run time checks (i.e. raising ValueError), as linters should detect any misspelt column args before it reaches production.

I made another commit with your suggestions and removed the runtime typos checking (raising `ValueError` for invalid column arguments). I still think it would be useful and if you think so too, I can revert it. If not, we can leave it as is and let linters to catch those typos.